pkg/agent/agent_linux.go

```diff
@@ -343,16 +343,30 @@ func (i *Initializer) installVMInitialFlows() error {
 // prepareL7NetworkPolicyInterfaces creates two OVS internal ports. An application-aware engine will connect to OVS
 // through these two ports.
 func (i *Initializer) prepareL7NetworkPolicyInterfaces() error {
-	trafficControlPortExternalIDs := map[string]interface{}{
-		interfacestore.AntreaInterfaceTypeKey: interfacestore.AntreaTrafficControl,
+	if err := i.prepareL7Port(config.L7NetworkPolicyTargetPortName); err != nil {
+		return err
 	}
+	if err := i.prepareL7Port(config.L7NetworkPolicyReturnPortName); err != nil {
+		return err
+	}
+	return nil
+}
 
-	for _, portName := range []string{config.L7NetworkPolicyTargetPortName, config.L7NetworkPolicyReturnPortName} {
-		_, exists := i.ifaceStore.GetInterface(portName)
-		if exists {
-			continue
-		}
+// prepareL7VisibilityInterfaces is required if featureGate l7Visibility is enabled but not l7NetworkPolicy.
```

Comment:
comment seems misleading to me. Based on the code above, `prepareL7VisibilityInterfaces` is called regardless of the status of L7NetworkPolicy.

Reply:
Taken care in the next comment